### PR TITLE
feat(fastly_alerts): add percentage alerts

### PR DESCRIPTION
**Pull request by jsonroy-fastly**
This pull requests add tests and documentation about how to use a percentage alerts.

### CODE REVIEW

File: fastly/resource_fastly_alert.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ func resourceFastlyAlert() *schema.Resource {
 				MaxItems:    1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"ignore_below": {
+							Type:        schema.TypeFloat,
+							Optional:    true,
+							Description: "Floor noise that can be configured to ignore data points that are below this threshold.",
```
Comment (Integralist) [inline fastly/resource_fastly_alert.go]: Could we rewrite this description to be more like the API docs? I personally find the API description makes more sense:
https://www.fastly.com/documentation/reference/api/observability/alerts/definition/#evaluation-strategy-data-model

> Threshold for the denominator value used in evaluations that calculate a rate or ratio. Usually used to filter out noise.

File: fastly/resource_fastly_alert.go [hunk 2]
```diff
@@ -291,11 +296,25 @@ func buildDimensions(data map[string][]string, v map[string]any) map[string][]st
...
+		evaluationStrategy["type"] = value.(string)
+	}
+
+	if value, ok := v["period"]; ok {
+		evaluationStrategy["period"] = value.(string)
 	}
+
+	if value, ok := v["threshold"]; ok {
+		evaluationStrategy["threshold"] = value.(float64)
+	}
```
Comment (Integralist) [inline fastly/resource_fastly_alert.go]: The key lookups in the map are redundant here as we specify those fields as `required` in the schema.

So I'd rework this like so:

```go
// Required attributes
m := map[string]any{
	"type":      v["type"].(string),
	"period":    v["period"].(string),
	"threshold": v["threshold"].(float64),
}
// Optional attributes
if value, ok := v["ignore_below"]; ok {
	evaluationStrategy["ignore_below"] = value.(float64)
}
return m
```

File: fastly/resource_fastly_alert_test.go [hunk 3]
```diff
@@ -193,6 +193,69 @@ func TestAccFastlyAlert_basic_stats_aggregate(t *testing.T) {
 	})
 }
 
+func TestAccFastlyAlert_basic_stats_aggregate_percent(t *testing.T) {
```
Comment (Integralist) [inline fastly/resource_fastly_alert_test.go]: We should really change all these test functions to remove the underscores...

e.g.
```suggestion
func TestAccFastlyAlertBasicStatsAggregatePercent(t *testing.T) {
```

**Review by Integralist — CHANGES_REQUESTED** [review]
LGTM. But there are some minor requested changes...

**Comment by Integralist** [comment]
/ok-to-test sha=330024a

**UPDATE:** The `validate-interface-fork` workflow (trigged by the above slash command) passed without error.

File: fastly/resource_fastly_alert.go (hunk 1 above)
Comment (jsonroy-fastly) [reply fastly/resource_fastly_alert.go]: Done

File: fastly/resource_fastly_alert.go (hunk 2 above)
Comment (jsonroy-fastly) [reply fastly/resource_fastly_alert.go]: Reworked as suggested

File: fastly/resource_fastly_alert_test.go (hunk 3 above)
Comment (jsonroy-fastly) [reply fastly/resource_fastly_alert_test.go]: I changed four of them to removed the `_`

**Comment by Integralist** [comment]
Sigh. Terraform.

So yeah it seems that Terraform has a thing where it expects one underscore between `TestWhatever_TheRestOfTheName` which is something to do with test name ambiguity.

Screenshot 2024-05-07 at 14 53 21

**Comment by Integralist** [comment]
https://pkg.go.dev/github.com/bflad/tfproviderlint@v0.29.0/passes/AT003

So I think we can just add a single underscore like so `TestAccFastlyAlert_TitleCaseTheRest`

File: fastly/resource_fastly_alert_test.go [hunk 4]
```diff
@@ -193,6 +193,69 @@ func TestAccFastlyAlert_basic_stats_aggregate(t *testing.T) {
...
+		Name:      gofastly.ToPointer(""),
+		ServiceID: gofastly.ToPointer(""),
+	}
+
+	createAlert := gofastly.AlertDefinition{
+		Description: "Terraform percent test",
+		Dimensions:  map[string][]string{},
+		// 25 percent increase
+		EvaluationStrategy: map[string]any{
+			"type":         "percent_increase",
```
Comment (darin-nee) [inline fastly/resource_fastly_alert_test.go:207]: Nice, I see we're adding tests for the new evaluation strategies, but the docs string for `type` doesn't call out these new options.

Comment (jsonroy-fastly) [reply fastly/resource_fastly_alert_test.go:207]: True, I added the missing ones, and the missing evaluation periods.

**Review by Integralist — APPROVED** [review]
LGTM. Did you have any other changes or is this ready to merge?

**Comment by jsonroy-fastly** [comment]
Ready to merge!

**Pull request merged**